### DISCUSSION TITLE
Does a CRON job use only 1 thread ?

### DISCUSSION THREAD

**Discussion by IoannisMaras** [OxygenFramework/Oxygen.jl#126] [General]
I was wondering if a cron job is using only 1 thread , because I run my oxygen app with 20 threads and I have a cron job that checks an array of model requests and if there is a request it handles the first one of a queue . When I run my script without oxygen it takes around 2-5 minutes but when I run it with oxygen (having 20 threads) it takes around 15-20 minutes.

If someone has handled anything similar please let me know 
Thank you in advance.

**Comment by ndortega** [comment]
Hi @IoannisMaras, 

After taking a look at the source code it appears you're absolutely correct, all cron jobs are run within an async task on the main thread of your Julia application regardless of how many threads are available. 

Thankfully, this should be a simple and quick fix. I should only have to change my `@async` call to a `Thread.@spawn` call when more than one thread is available. This will ensure it schedules it on any available thread so it doesn't slow down the main thread. 

With that said, what kind of hardware are you running your web application on? Running with 20 threads is the highest number I've heard from anyone using this package. The reason I bring this up is that adding more threads doesn't always make the performance better. Once you get past a certain threshold it actually negatively impacts performance and will slow down your api by a lot. Anecdotally, my personal computer has 12 cores and I generally get the best performance when running with 2-4 threads

**Reply by IoannisMaras** [reply]
Thank you very much.

This app is used to serve a lot of requests in my workspace and it's hosted on Azure . That's why It has 20 threads and will possibly have more in the future. You think I will run into a problem with this ?
